app/gui2/src/util/events.ts

```diff
@@ -31,6 +51,24 @@ export function useWindowEvent<K extends keyof WindowEventMap>(
   })
 }
 
+/**
+ * Add an event listener on document for the duration of component lifetime.
+ * @param event name of event to register
+ * @param handler event handler
+ */
+export function useDocumentEvent<K extends keyof DocumentEventMap>(
+  event: K,
+  handler: (e: DocumentEventMap[K]) => void,
+  options?: boolean | AddEventListenerOptions,
+): void {
+  onMounted(() => {
```

Comment:
this could now use `useElementEvent`.

Reply:
`document` isn't an `Element` though (however it is a `Node`): [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document) - so i'm pretty sure its list of valid events is slightly different

Reply:
Right. Ok let's keep it separate then.